File: DailyProgram.java
import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class DailyProgram {
    private Date day; // Represents the specific day
    private Map<Activity, List<Student>> activities; // Maps activities to their participants

    public DailyProgram(Date day) {
        this.day = day;
        this.activities = new HashMap<>();
    }

    // Getter and Setter for the day
    public Date getDay() {
        return day;
    }

    public void setDay(Date day) {
        this.day = day;
    }

    // Add an activity with participants
    public void addActivity(Activity activity, List<Student> participants) {
        activities.put(activity, participants);
    }

    // Get all activities
    public Map<Activity, List<Student>> getActivities() {
        return activities;
    }

    // Get participants for a specific activity
    public List<Student> getParticipants(Activity activity) {
        return activities.get(activity);
    }

    // Remove an activity
    public void removeActivity(Activity activity) {
        activities.remove(activity);
    }

    public void saveToDatabase() {
        try (Connection connection = DbConnection.connect()) {
            String sql = "INSERT INTO daily_program (day, activity_id) VALUES (?, ?)";
            for (Map.Entry<Activity, List<Student>> entry : activities.entrySet()) {
                PreparedStatement statement = connection.prepareStatement(sql);
                statement.setDate(1, new java.sql.Date(day.getTime()));
                statement.setInt(2, entry.getKey().getId());
                statement.executeUpdate();
            }
            System.out.println("Daily program saved to database.");
        } catch (SQLException e) {
            System.err.println("Error saving daily program to database: " + e.getMessage());
        }
    }

    @Override
    public String toString() {
        return "DailyProgram [day=" + day + ", activities=" + activities + "]";
    }
}

File: old/daily_program.py
from PythonExpenseApp.db_connection import DbConnection  # Importa la classe per la connessione al database
import mysql.connector  # Importa il modulo per la connessione a MySQL

class DailyProgram:
    def __init__(self, day):
        self.day = day  # Giorno del programma (può essere un oggetto datetime.date o simile)
        self.activities = {}  # Dizionario: chiave = oggetto Activity, valore = lista di Student

    def add_activity(self, activity, participants):
        self.activities[activity] = participants  # Aggiunge un'attività e la lista dei partecipanti al dizionario

    def get_activities(self):
        return self.activities  # Restituisce tutte le attività del giorno

    def get_participants(self, activity):
        return self.activities.get(activity, [])  # Restituisce la lista dei partecipanti per una data attività

    def remove_activity(self, activity):
        if activity in self.activities:
            del self.activities[activity]  # Rimuove un'attività dal programma del giorno

    def save_to_database(self):
        connection = DbConnection.connect()  # Ottiene la connessione al database
        if connection:
            try:
                sql = "INSERT INTO daily_program (day, activity_id) VALUES (%s, %s)"  # Query per salvare il programma giornaliero
                statement = connection.cursor()  # Crea un cursore per eseguire la query
                for activity, _ in self.activities.items():  # Per ogni attività nel programma
                    statement.execute(sql, (self.day, activity.get_id()))  # Salva giorno e id attività nel database
                connection.commit()  # Conferma le modifiche
                print("Daily program saved to database.")  # Messaggio di conferma
            except mysql.connector.Error as e:  # Gestione errori MySQL
                print(f"Error saving daily program to database: {e}")

    def __str__(self):
        return f"DailyProgram [day={self.day}, activities={self.activities}]"  # Rappresentazione testuale dell'oggetto